sys/bitfield/bitfield.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_util
+ * @{
+ *
+ * @file
+ * @brief       Bitfield auxillary functions
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include "bitfield.h"
+#include "irq.h"
+
+int bf_get_unset(uint8_t field[], int size)
+{
+    unsigned state = disableIRQ();
```

Comment:
moved disableIRQ down. Using a mutex makes this otherwise stateless function need a state struct, initialization, runtime memory... Let's postpone evaluation of both options.